src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -52,11 +57,23 @@ public class TagHelperContext
         /// This <see cref="IDictionary{Object, Object}" /> is copy-on-write in order to ensure items added to this
         /// collection are visible only to other <see cref="ITagHelper"/>s targeting child elements.
         /// </remarks>
-        public IDictionary<object, object> Items { get; }
+        public IDictionary<object, object> Items { get; private set; }
 
         /// <summary>
         /// An identifier unique to the HTML element this context is for.
         /// </summary>
-        public string UniqueId { get; }
+        public string UniqueId { get; private set; }
+
+        /// <summary>
+        /// Clears the <see cref="TagHelperContext"/> and updates its state with the provided values.
+        /// </summary>
+        /// <param name="items">The <see cref="IDictionary{Object, Object}"/> to use.</param>
+        /// <param name="uniqueId">The unique id to use.</param>
+        public void Reinitialize(IDictionary<object, object> items, string uniqueId)
```

Comment:
Why use the name `Reinitialize()` here, in `TagHelperExecutionContext` and in `TagHelperOutput` but `Reset()` in `TagHelperContent`? Think `Reinitialize()` works everywhere.